crates/solidity/testing/snapshots/bindings_output/built_ins/shadowing/generated/0.7.1-success.txt

```diff
@@ -21,10 +21,10 @@ References and definitions:
     │           ╰──── ref: 4
   7 │         tx;
     │         ─┬  
-    │          ╰── ref: 2
+    │          ╰── refs: 2, built-in
   8 │         sha256();
```

Comment:
Looks like all functions with the same name are returned now, rather than resolving the closest one. Is this expected?

Reply:
This was previously resolved with the ranking algorithm applied to all results we got from the stack graph. For this case, built-ins were being de-prioritized by ranking. In all other cases, those are either virtual methods which were ranked using C3 linearisation, or import aliases.

Reply:
Returning multiple method overloads for a method call is useful for many scenarios, and they are all correct. If an IDE does a rename, they all should be renamed. We consider them the same one.
But not sure this reasoning applies to this case. Here, the built-in is shadowed by the contract function. They are not the same definition. They are two separate ones that can be renamed/moved separately.

I assume this would have been handled by scopes, rather than ranking?

1. The global scope has access to the built-in `sha256` function.
2. The outer scope of the file adds the global `sha256` function.
3. The inner scope of the contract adds the private `sha256` function.
4. Name resolution of the `sha256()` method call inside `test()` should pick #3, and stop there. Right?

Why is it different than shadowed variables, or how locally-defined variables should shadow variables brought in from an `import * from "...."` statement?

Reply:
Yeah, you're right. This case should be handled by scoping. We're missing setting  precedence in edges for in contract definitions. I'll fix this.